packages/next/src/build/index.ts

```diff
@@ -870,6 +870,19 @@ export default async function build(
           )
         )
 
+      // We need to write a partial prerender manifest to make preview mode settings available in edge middleware
+      const partialManifest: Partial<PrerenderManifest> = {
+        preview: previewProps,
+      }
+
+      await fs.writeFile(
+        path.join(distDir, PRERENDER_MANIFEST).replace(/\.json$/, '.js'),
+        `self.__PRERENDER_MANIFEST=${JSON.stringify(
+          JSON.stringify(partialManifest)
+        )}`,
```

Comment:
I think theres a bug since you have two nested `JSON.stringify()`

Reply:
ty -- this was copied from [here](https://github.com/vercel/next.js/blob/canary/packages/next/src/build/index.ts#L3071-L3077) so I'll update both

Reply:
@styfle turns out this was intentional -- the first stringify is to convert into a JSON string and the second stringify for escaping

eg:
```js
const foo = { key: "value" }
const bar = JSON.stringify(foo); // '{"key":"value"}'
const foobar = JSON.stringify(bar); // '"{\\"key\\":\\"value\\"}"'
```